projects/packages/my-jetpack/_inc/components/plans-section/index.jsx

```diff
@@ -177,13 +177,17 @@ function PlanSectionFooter( { numberOfPurchases } ) {
  */
 export default function PlansSection() {
 	const userIsAdmin = !! getMyJetpackWindowInitialState( 'userIsAdmin' );
+	const { isSiteConnected } = useMyJetpackConnection();
+
 	const {
 		data: purchases,
 		isLoading,
 		isError,
 	} = useSimpleQuery( {
 		name: QUERY_PURCHASES_KEY,
-		query: { path: REST_API_SITE_PURCHASES_ENDPOINT },
+		query: {
+			path: isSiteConnected ? REST_API_SITE_PURCHASES_ENDPOINT : null,
```

Comment:
Not sure if it's a common practice in this codebase already. If it is, feel free to ignore this comment.

But I thought a little weird to see the query with a `null` path. I see it is sent to the `apiFetch`, and it's the part where it's probably skipped?

If it makes sense to you, I was thinking we could implement it in a different way. I think if we pass the `options: { enabled: isSiteConnected }`. WDYT?

Reply:
Changed